packages/cozy-pouch-link/src/utils.js

```diff
@@ -27,3 +27,47 @@ export const formatAggregatedError = aggregatedError => {
 
   return strings.join('\n')
 }
+
+/**
+ * @typedef {object} FulfilledPromise
+ * @property {'fulfilled'} status - The status of the promise
+ * @property {undefined} reason - The Error rejected by the promise (undefined when fulfilled)
+ * @property {any} value - The resolved value of the promise
+ */
+
+/**
+ * @typedef {object} RejectedPromise
+ * @property {'rejected'} status - The status of the promise
+ * @property {Error} reason - The Error rejected by the promise
+ * @property {undefined} value - The resolved value of the promise (undefined when rejected)
+ */
+
+/**
+ * Takes an iterable of promises as input and returns a single Promise.
+ * This returned promise fulfills when all of the input's promises settle (including
+ * when an empty iterable is passed), with an array of objects that describe the
+ * outcome of each promise.
+ * This implementation is useful for env with no support of the native Promise.allSettled,
+ * typically react-native 0.66
```

Comment:
According to https://github.com/facebook/react-native/releases/tag/v0.70.6, we can now safely use Promise.allSettled on the flagship app.

Reply:
But I assume we might have users using out-of-date flagship in RN 0.66?

Reply:
Maybe, not prioritary to do anyway, but I think it starts to be safe to change

Reply:
This won't be problematic as this is packaged in the build, so old flagship version will use the old code with custom method and new versions will use the new one. No retro-compatibility needed.

Reply:
But there is no priority at all as those methods are equivalent